pkg/controller/storagecluster/cephcluster.go

```diff
@@ -166,6 +169,12 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, nodeCount int, r
 		"app": sc.Name,
 	}
 
+	// determining the kube server version
+	serverVersion, err := determineKubeVersion()
+	if err != nil {
+		reqLogger.Error(err, "failed to determine the kube server version.")
```

Comment:
I agree on returning an error and stop the flow, as the user will be unaware with the branching that it takes.
Since we are returning an error in case of failure detection, faced difficulties in the existing unit tests to succeed (unable to fake the serverVersion retrieval). As a better approach, have integrated the serverVersion field in `ReconcileStorageCluster struct` and modified accordingly.